controllers/common/helper.go

```diff
@@ -665,8 +665,6 @@ func HandleUpstreamClusterCleanup(
 			}
 			return false, err
 		}
```

Comment:
LGTM, but one suggestion that's unrelated to the changes in this PR. 'm wondering if the code above this can be simplified:

I see there's a Get and then Delete. The Get seems redundant. Why not just do a Delete? If success, then return false (retry). If NotFoundErr, then remove finalizer and update

Reply:
@mayankshah1607 I think the Get is there bc of how the `common.HandleUpstreamClusterCleanup` is being called from the provider: 
https://github.com/percona/everest-operator/blob/cbfab94d6cff424e51c98561af53bff749209d97/controllers/providers/psmdb/provider.go#L170
So provider receives the technology-specific empty object and then gets the actual object from k8s to delete it. Otherwise the logic that gets the upstream CR would need to be duplicated in each `Cleanup()` implementation.